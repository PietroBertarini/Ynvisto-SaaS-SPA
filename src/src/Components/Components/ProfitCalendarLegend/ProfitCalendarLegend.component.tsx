import React from 'react';
import clsx from 'clsx';
import { Typography } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import { useStyles } from './ProfitCalendarLegend.styles';

export interface OwnProps {
    /**
     * Actual Month Profit and Taxes Values.
     */
    legendData : ILegendData[];
}
export interface ILegendData {
    /**
     * Choose a color of the helper.
     */
    color : EColor;
    /**
     * The label of the helper
     */
    label : string;
}

export enum EColor {
    RED = 'RED',
    BLUE = 'BLUE',
    GREEN = 'GREEN',
    YELLOW = 'YELLOW',
    WHITE = 'WHITE',
}

/**
 * Actual Month Profit
 */
function ProfitCalendarLegend(props: OwnProps) {
  const {
    legendData,
  } = props;

  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Box display="flex" flexDirection="row">
        {legendData.map((item : ILegendData) => (
          <FormRow key={item.label} color={item.color} label={item.label} />
        ))}
      </Box>
    </div>
  );
}

export interface FormRowProps {
    color : EColor;
    key: string;
    label: string;
}
function FormRow(props: FormRowProps) {
  const { label } = props;
  const classes = useStyles(props);

  const circle = <div className={clsx(classes.shape, classes.shapeCircle)} />;

  return (
    <Box display="flex" flexDirection="row" justifyContent="space-between" alignItems="center" className={classes.row}>
      {circle}
      <Typography className={classes.subTitle}>
        {label}
      </Typography>
    </Box>
  );
}
export default ProfitCalendarLegend;
