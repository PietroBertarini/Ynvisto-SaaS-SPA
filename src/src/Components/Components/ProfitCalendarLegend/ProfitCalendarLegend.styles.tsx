import { createStyles, makeStyles } from '@material-ui/core/styles';

const styledBy = (property, mapping) => (props) => mapping[props[property]];
export const useStyles = makeStyles(() => createStyles({
  root: {
    paddingBottom: '1.2vw',
  },
  row: {
    marginLeft: '4vw',
  },
  shape: {
    backgroundColor: styledBy('color', {
      BLUE: '#05C0E9',
      GREEN: '#05E985',
      YELLOW: '#E9C46A',
      RED: '#EB5050',
      WHITE: '#B1B1B1',
    }),
    // /
    width: '1.2vw',
    height: '1.2vw',
    marginRight: '0.6vw',
  },
  shapeCircle: {
    borderRadius: '50%',
  },
  subTitle: {
    color: '#FFFFFF',
    fontSize: '1vw',
  },
}));
