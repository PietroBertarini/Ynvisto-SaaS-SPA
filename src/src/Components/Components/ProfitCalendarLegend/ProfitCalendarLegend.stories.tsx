import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import ProfitCalendarLegend, { OwnProps } from './ProfitCalendarLegend.component';
import { mockCalendarLegend } from '../../../Mocks/ProfitCalendarLegend.mock';

export default {
  title: 'Components/ProfitCalendarLegend',
  component: ProfitCalendarLegend,
} as Meta;

const Template: Story<OwnProps> = (args) => <ProfitCalendarLegend {...args} />;

export const ProfitCalendarHelperDefault = Template.bind({});
ProfitCalendarHelperDefault.args = {
  legendData: mockCalendarLegend,
};
