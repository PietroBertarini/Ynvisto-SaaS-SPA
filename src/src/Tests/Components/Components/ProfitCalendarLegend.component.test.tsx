import React from 'react';
import { shallow } from 'enzyme';
import ProfitCalendarLegend from '../../../Components/Components/ProfitCalendarLegend/ProfitCalendarLegend.component';
import { mockCalendarLegend } from '../../../Mocks/ProfitCalendarLegend.mock';

test('Renders ProfitCalendarLegend component and match to previously snapshot', () => {
  const component = shallow(<ProfitCalendarLegend legendData={mockCalendarLegend} />);
  expect(component).toMatchSnapshot();
});
