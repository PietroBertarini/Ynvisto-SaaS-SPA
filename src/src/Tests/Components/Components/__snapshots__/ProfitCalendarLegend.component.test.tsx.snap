// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ProfitCalendarLegend component and match to previously snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <Styled(MuiBox)
    display="flex"
    flexDirection="row"
  >
    <FormRow
      color="BLUE"
      key="DARF Paga"
      label="DARF Paga"
    />
    <FormRow
      color="GREEN"
      key="Não houve DARF"
      label="Não houve DARF"
    />
    <FormRow
      color="YELLOW"
      key="DARF Pendente"
      label="DARF Pendente"
    />
    <FormRow
      color="RED"
      key="DARF Atrasada"
      label="DARF Atrasada"
    />
    <FormRow
      color="WHITE"
      key="Mês não finalizado"
      label="Mês não finalizado"
    />
  </Styled(MuiBox)>
</div>
`;
